pkg/mqtrigger/mqtrigger.go

```diff
@@ -40,10 +44,68 @@ func Start(logger *zap.Logger, routerUrl string) error {
 	// Message queue type: nats is the only supported one for now
 	mqType := os.Getenv("MESSAGE_QUEUE_TYPE")
 	mqUrl := os.Getenv("MESSAGE_QUEUE_URL")
+
+	// For authentication with message queue
+	secrets, err := readSecrets(logger)
+
+	if err != nil {
+		panic(err)
+	}
+
 	mqCfg := messageQueue.MessageQueueConfig{
-		MQType: mqType,
-		Url:    mqUrl,
+		MQType:  mqType,
+		Url:     mqUrl,
+		Secrets: secrets,
 	}
 	messageQueue.MakeMessageQueueTriggerManager(logger, fissionClient, routerUrl, mqCfg)
 	return nil
 }
+
+func getCurrentNamespace() string {
+	data, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+
+	namespace := strings.TrimSpace(string(data))
+	if err == nil && len(namespace) > 0 {
+		return namespace
+	}
+
+	return "default"
+}
+
+func readSecrets(logger *zap.Logger) (map[string][]byte, error) {
+	secretsPath := "/etc/secrets"
+	secrets := make(map[string][]byte)
+
+	// return if no secrets exist
+	if _, err := os.Stat(secretsPath); os.IsNotExist(err) {
+		return secrets, nil
```

Comment:
`return nil, err`
I personally prefer to return `nil` for the fields other than error.

Reply:
Done